Source/Core/DolphinQt/Achievements/AchievementHeaderWidget.cpp

```diff
@@ -76,38 +74,93 @@ void AchievementHeaderWidget::UpdateData()
 {
   if (!AchievementManager::GetInstance()->IsLoggedIn())
   {
-    m_user_box->setVisible(false);
-    m_game_box->setVisible(false);
+    m_header_box->setVisible(false);
     return;
   }
 
-  QString user_name =
-      QString::fromStdString(AchievementManager::GetInstance()->GetPlayerDisplayName());
-  m_user_name->setText(user_name);
-  m_user_points->setText(tr("%1 points").arg(AchievementManager::GetInstance()->GetPlayerScore()));
+  AchievementManager::PointSpread point_spread;
+  QString user_name, game_name;
+  AchievementManager::BadgeStatus player_badge;
+  AchievementManager::BadgeStatus game_badge;
+  {
+    AchievementManager::GetInstance()->GetLock();
```

Comment:
Uh, this line doesn't do anything, you need to actually lock the mutex.

Reply:
Okay, so I can't just use the same getlock as everywhere in AchievementManager. Do you have a recommendation for how Dolphin would usually do this?